ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouterFactory.java

```diff
@@ -93,7 +104,7 @@ public NonBlockingRouterFactory(VerifiableProperties verifiableProperties, Clust
   public Router getRouter() throws InstantiationException {
     try {
```

Comment:
I wonder if there's a way to test that the KMS and CryptoService classes match key types before startup succeeds, ideally not by performing an actual decrypt job. Perhaps, each service can expose a `Class<T> getSupportedKeyClass()` method and you can use `Class.isAssignableFrom(...)` to check.

Reply:
Good point. Can you create an issue. we can work on it later.